src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IChecksummedPersistentStorage.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal interface IChecksummedPersistentStorage : IPersistentStorage
+    {
+        /// <summary>
+        /// Reads the existing checksum we have for the solution with the given <paramref name="name"/>,
+        /// or <see langword="null"/> if we do not have a checksum persisted.
+        /// </summary>
+        Task<Checksum> ReadChecksumAsync(string name, CancellationToken cancellationToken = default);
```

Comment:
Note: instead of a Checksum, we could consider just using a Sha1Hash. It's just a simple struct that Checksum itself wraps.  that would enable us to get rid of an unnecessary allocation on these paths.